src/plugins/discover/public/components/discover_grid/get_render_cell_value.tsx

```diff
@@ -64,89 +63,35 @@ export const getRenderCellValueFn =
-      const highlightPairs: Array<[string, string]> = [];
-      const sourcePairs: Array<[string, string]> = [];
-      Object.entries(innerColumns).forEach(([key, values]) => {
-        const subField = indexPattern.getFieldByName(key);
-        const displayKey = indexPattern.fields.getByName
-          ? indexPattern.fields.getByName(key)?.displayName
-          : undefined;
-        const formatter = subField
-          ? indexPattern.getFormatterForField(subField)
-          : { convert: (v: unknown, ...rest: unknown[]) => String(v) };
-        const formatted = (values as unknown[])
-          .map((val: unknown) =>
-            formatter.convert(val, 'html', {
-              field: subField,
-              hit: row,
-              indexPattern,
-            })
-          )
-          .join(', ');
-        const pairs = highlights[key] ? highlightPairs : sourcePairs;
-        if (displayKey) {
-          if (fieldsToShow.includes(displayKey)) {
-            pairs.push([displayKey, formatted]);
-          }
-        } else {
-          pairs.push([key, formatted]);
-        }
-      });
-
-      return (
-        // If you change the styling of this list (specifically something that will change the line-height)
-        // make sure to adjust the img overwrites attached to dscDiscoverGrid__descriptionListDescription
-        // in discover_grid.scss
-        <EuiDescriptionList type="inline" compressed className="dscDiscoverGrid__descriptionList">
-          {[...highlightPairs, ...sourcePairs]
-            .slice(0, maxDocFieldsDisplayed)
-            .map(([key, value]) => (
-              <Fragment key={key}>
-                <EuiDescriptionListTitle>{key}</EuiDescriptionListTitle>
-                <EuiDescriptionListDescription
-                  dangerouslySetInnerHTML={{ __html: value }}
-                  className="dscDiscoverGrid__descriptionListDescription"
-                />
-              </Fragment>
-            ))}
-        </EuiDescriptionList>
-      );
-    }
+        !field &&
+        row?.fields &&
+        !(row.fields as Record<string, unknown[]>)[columnId]
+    );
 
-    if (typeof rowFlattened[columnId] === 'object' && isDetails) {
-      return (
-        <JsonCodeEditor
-          json={rowFlattened[columnId] as Record<string, unknown>}
-          width={defaultMonacoEditorWidth}
-        />
+    if (isDetails) {
+      return renderPopoverContent(
+        row,
+        rowFlattened,
+        field,
+        columnId,
+        indexPattern,
+        useTopLevelObjectColumns
       );
     }
 
-    if (field && field.type === '_source') {
-      if (isDetails) {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        return <JsonCodeEditor json={row as any} width={defaultMonacoEditorWidth} />;
-      }
-      const pairs = formatHit(row, indexPattern, fieldsToShow);
+    if (field?.type === '_source' || useTopLevelObjectColumns) {
```

Comment:
Does it make sense to create separate variable for this check? like you did for `useTopLevelObjectColumns`

Reply:
I'm struggling here finding a good name that would describe it, do you have suggestion?
like `renderPrettyJSON`?